tools/ipfs-deploy.js

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env node
+
+/**
+ * Run on Github by commits to master, this script:
+ * 1. Gets all projects with a fleek configuration file
+ *   2. Gets the last commit for the relevant site id
+ *   3. Runs nx:affected for that commit ID and checks if the site has changed
+ *     4. Calls deploy if it has
+ *
+ * This would probably be best as an NX task, but the circular nature of getting
+ * the fleek ID for the project, then checking if it was affected didn't fit within the
+ * build-only-affected cycle, and as each fleek deploy will have been triggered by
+ * a different commit, it seemed best to do this outwith nx. Feel free to re-implement
+ * this if the assumptions are wrong.
+ *
+ * It has also been written to skip external dependencies.
+ */
+const { existsSync, readdirSync, lstatSync, readFileSync } = require('fs');
+const { execSync } = require('child_process');
+
+/**
+ * Parses the last commit hash out of the Fleek API response
+ * @param {String} siteId
+ * @returns string Last commit that triggered a build
+ */
+function getFleekLastBuildCommit(siteId) {
+  const curl = `curl 'https://api.fleek.co/graphql' --silent -X POST -H 'Accept: */*' -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Authorization: ${process.env['FLEEK_API_KEY']}' --data-raw '{"query":"{getSiteById(siteId: \\"${siteId}\\"){publishedDeploy{repository{commit}}}}","variables":null}'`;
```

Comment:
Thanks got there will be fetch API available soon
https://nodejs.org/en/blog/release/v17.5.0/

Reply:
Hehe yeah. `require('https')` was so heavyweight. But more sensible. But... so inelegant. I expected you to reject the review based on this 😆